### ISSUE TITLE
Different accuracy when predicting the dev set with do_predict

### ISSUE THREAD

**Issue by atifemreyuksel** [google-research/bert#452]
I train my dataset(the same format with CoLA dataset) for sentence classification model and I take the evaluation results in eval mode after training which is 
eval_accuracy = 0.79638886
eval_loss = 1.4008523
global_step = 128711
loss = 1.4008523

Afterthat, I copied this dev.tsv into test.tsv by removing the labels and the other columns in order to construct the same format of test.csv of CoLa. After I took the prediction results, I compare the predictions with their own labels. When I calculate the accuracy ((TP + TN) / (TP + TN + FP + FN)), the accuracy is equal to 0.48138888888888887. 

In brief, the accuracy scores are different when I predict the same dataset in eval mode and as test.csv predict by using do_predict option. Is it possible? Thanks for all comments.

**Comment by hainan89** [comment]
I also got the similar issue.

I notice that in the run_classifier.py script, when read test data, the label also should be assigned. (eg. line 288, 328, 368). It really makes me confused.

Generally, when we do prediction or test, there is no labels should be given.

When training my own dataset, everything goes well. But, when the same dataset was used to do the test, following the run_classifier.py script, I also give a constant value to all the test data. The test results all are the given constant values.

@atifemreyuksel  do you have any progress？

**Comment by atifemreyuksel** [comment]
Which data processor do you use? I am using CoLA data processor and I don't give the labels of the instances in test dataset. I construct test.tsv by giving index and sentence as columns. 

In lines 288, 328, 368, the unnecessary label is assigned to test set, you are right. But this label is not used in evaluation when you want to take predictions by using --do_predict option. 

I can take the right predictions of the same dataset in test.tsv by using the [instruction of taking predictions](https://github.com/google-research/bert#prediction-from-classifier). After you take the test_results.tsv, you can see the softmax results of each instances. You can find the maximum index of each row in order to reach the prediction of the model.

**Comment by hainan89** [comment]
Hi @atifemreyuksel , thanks for your kind reminder. I got the errors, it is not the issue of the model. It is because when doing the test, I still use the unnecessary label attached in the InputFeatures to calculate the prediction accuracy. That is why I got a very low accuracy. Really a stupied mistake. ╮(╯▽╰)╭

**Comment by NancyLele** [comment]
@hainan89  
i met the same problem。If the test data has no label, then program  interruption.
Code below:
         if set_type == "test":
                text_a = tokenization.convert_to_unicode(line[0])
                examples.append(
                    InputExample(guid=guid, text_a=text_a, text_b=None,label=None))
The error is file_based_converter_to example return none

**Comment by oakkas** [comment]
I had the same problem and figured out that, if you give separate output dir while only do eval or test, it doesn't find the model dir. Since;
run_config = tf.contrib.tpu.RunConfig(
      cluster=tpu_cluster_resolver,
      master=FLAGS.master,
      model_dir=FLAGS.output_dir,
      save_checkpoints_steps=FLAGS.save_checkpoints_steps,
      tpu_config=tf.contrib.tpu.TPUConfig(
          iterations_per_loop=FLAGS.iterations_per_loop,
          num_shards=FLAGS.num_tpu_cores,
          per_host_input_for_training=is_per_host))
above model_dir set to output_dir and if you give a separate dir (i.e. give a dir to output eval/test results), it does't produce same results. 
So you either have it save eval/test to the same training/model folder or manually set the model_dir from above code.
